File: skycast/src/main/java/com/skycast/model/Weather.java
package com.skycast.model;

import jakarta.persistence.*;
import lombok.Data;

import java.math.BigDecimal;

@Entity
@Table(name = "weather")
@Data
public class Weather {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "dt")
    private String dt;

    @Column(name = "weather_description")
    private String weatherDescription;

    @Column(name = "weather_icon")
    private String weatherIcon;

    @Column(name = "temp")
    private BigDecimal temp;

    @Column(name = "feels_like")
    private BigDecimal feelsLike;

    @Column(name = "temp_min")
    private BigDecimal tempMin;

    @Column(name = "temp_max")
    private BigDecimal tempMax;

    @Column(name = "pressure")
    private Integer pressure;

    @Column(name = "humidity")
    private Integer humidity;

    @Column(name = "wind_speed")
    private BigDecimal windSpeed;

    @Column(name = "wind_deg")
    private Integer windDeg;

    @Column(name = "clouds_all")
    private Integer cloudsAll;

    @Column(name = "sys_sunrise")
    private String sysSunrise;

    @Column(name = "sys_sunset")
    private String sysSunset;

    @ManyToOne
    @JoinColumn(name = "city_id", referencedColumnName = "id")
    private City city;

}

File: skycast/skycast-app/src/app/weather.ts
import { City } from "./city";

export interface Weather {

    id: number;
    dt: string;
    weatherDescription: string;
    weatherIcon: string;
    temp: number;
    feelsLike: number;
    tempMin: number;
    tempMax: number;
    pressure: number;
    humidity: number;
    windSpeed: number;
    windDeg: number;
    cloudsAll: number;
    sysSunrise: string;
    sysSunset: string;
    city: City;

}